x-pack/plugins/apm/public/components/app/service_overview/service_overview_charts/most_used_chart/most_used_chart.test.tsx

```diff
@@ -0,0 +1,156 @@
+ * 2.0.
+ */
+
+import { render } from '@testing-library/react';
+import { MemoryRouter } from 'react-router-dom';
+import { CoreStart } from '@kbn/core/public';
+import React, { ReactNode } from 'react';
+import { createKibanaReactContext } from '@kbn/kibana-react-plugin/public';
+import { MockApmPluginContextWrapper } from '../../../../../context/apm_plugin/mock_apm_plugin_context';
+import { getLensAttributes } from './get_lens_attributes';
+import { MostUsedChart, MostUsedMetricTypes } from '.';
+import { HOST_OS_VERSION } from '../../../../../../common/elasticsearch_fieldnames';
+
+const mockEmbeddableComponent = jest.fn();
+
+function Wrapper({ children }: { children?: ReactNode }) {
+  const KibanaReactContext = createKibanaReactContext({
+    lens: {
+      EmbeddableComponent: mockEmbeddableComponent.mockReturnValue(
+        <div data-test-subj="lens-mock" />
+      ),
+      canUseEditor: jest.fn(() => true),
+      navigateToPrefilledEditor: jest.fn(),
+    },
+  } as Partial<CoreStart>);
+
+  return (
+    <MemoryRouter>
+      <KibanaReactContext.Provider>
+        <MockApmPluginContextWrapper>{children}</MockApmPluginContextWrapper>
+      </KibanaReactContext.Provider>
+    </MemoryRouter>
+  );
+}
+
+const renderOptions = { wrapper: Wrapper };
+
+describe('Most used chart with Lens', () => {
+  test('gets lens attributes', () => {
+    const props = {
+      metric: HOST_OS_VERSION as MostUsedMetricTypes,
+      filters: [
+        {
+          meta: {},
+          query: {
+            term: {
+              'processor.event': 'transaction',
+            },
+          },
+        },
+        {
+          meta: {},
+          query: {
+            term: {
+              'service.name': 'opbeans-swift',
+            },
+          },
+        },
+        {
+          meta: {},
+          query: {
+            term: {
+              'transaction.type': 'request',
+            },
+          },
+        },
+      ],
+    };
+
+    expect(getLensAttributes(props)).toMatchSnapshot();
+  });
+
+  test('Renders most used chart with Lens', () => {
+    const start = '2022-10-30T20%3A52%3A47.080Z';
+    const end = '2022-10-31T20%3A52%3A47.080Z';
+    const props = {
+      metric: HOST_OS_VERSION as MostUsedMetricTypes,
+      filters: [
+        {
+          meta: {},
+          query: {
+            term: {
+              'processor.event': 'transaction',
+            },
+          },
+        },
+        {
+          meta: {},
+          query: {
+            term: {
+              'service.name': 'opbeans-swift',
+            },
+          },
+        },
+        {
+          meta: {},
+          query: {
+            term: {
+              'transaction.type': 'request',
+            },
+          },
+        },
+      ],
+    };
+    render(
+      <MostUsedChart
+        title="Most used os version"
+        start={start}
+        end={end}
+        metric={HOST_OS_VERSION as MostUsedMetricTypes}
+        filters={[
```

Comment:
Filters are also used above in `props` so they can be defined as a variable and reused.

Reply:
https://github.com/elastic/kibana/pull/144232/commits/25c2b7c0078c7029e29f8b1b1ab47a816969b2e6